Upgrade segment not cleared after an upgrade starting from 24.0.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

After the upgrade to v24.0.0 and the cache_timeout removal, when the upgrade segment indicates that an update is available and when you install the new version the segment is not removed until you restart your shell. This was not the case previously with the cache_timeout mechanism. Can this be improved?

### Theme

[mytheme.omp.json](https://github.com/user-attachments/files/17649219/mytheme.omp.json)


### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 24.0.8

Shell: bash (5.2.32(1)-release)

Prompt:

╭─   pascal   \[\\]󱙴 /git\[\\]  \[\\] \[\\] master ≡  ?1                                                                                                                                                                              97   56.29% 
╰─ 

Segments:

ConsoleTitle(true)                         -   0 ms
Os(true)                                   -   0 ms
Session(true)                              -   2 ms
Upgrade(false)                             - 102 ms
Path(true)                                 -   2 ms
Git(true)                                  -  32 ms
Svn(false)                                 -   0 ms
Root(false)                                -   0 ms
Status(true)                               -   0 ms
Executiontime(false)                       -   0 ms
Battery(true)                              -   0 ms
Sysinfo(true)                              -   1 ms
Text(true)                                 -   0 ms
Status(true)                               -   0 ms

Run duration: 111.851674ms

Cache path: /home/pascal/.cache/oh-my-posh

Config path: /home/pascal/.config/mytheme.omp.json

Logs:

[DEBUG] 16:25:19.418 debug.go:49 → debug mode enabled
[DEBUG] 16:25:19.418 debug.go:49 → plain mode enabled
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(LOCALAPPDATA) - 42.672µs
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(OMP_CACHE_DIR) - 4.571µs
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(XDG_CACHE_HOME) - 8.329µs
[TRACE] 16:25:19.418 terminal.go:CachePath() - 105.723µs
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache file: /home/pascal/.cache/oh-my-posh/omp.cache
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache key: is_wsl
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache key: notice
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache key: upgrade_check
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache key: environment_platform
[TRACE] 16:25:19.418 terminal.go(/home/pascal/.cache/oh-my-posh/omp.cache) - 150.397µs
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(LOCALAPPDATA) - 4.711µs
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(OMP_CACHE_DIR) - 4.06µs
[DEBUG] 16:25:19.418 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.418 terminal.go:Getenv(XDG_CACHE_HOME) - 3.877µs
[TRACE] 16:25:19.418 terminal.go:CachePath() - 47.489µs
[DEBUG] 16:25:19.418 terminal.go:65 → loading cache file: /home/pascal/.cache/oh-my-posh/omp.cache.bc2a6813-be22-4e3c-a014-14bab9172ddc
[DEBUG] 16:25:19.419 terminal.go:65 → loading cache key: prompt_count_cache
[DEBUG] 16:25:19.419 terminal.go:65 → loading cache key: template_cache
[TRACE] 16:25:19.419 terminal.go(/home/pascal/.cache/oh-my-posh/omp.cache.bc2a6813-be22-4e3c-a014-14bab9172ddc) - 191.897µs
[TRACE] 16:25:19.419 terminal.go:setPromptCount() - 837ns
[TRACE] 16:25:19.419 terminal.go:GOOS() - 117ns
[DEBUG] 16:25:19.419 terminal.go:setPwd:194 → /home/pascal/tools/wireshark/git
[TRACE] 16:25:19.419 terminal.go:setPwd() - 35.341µs
[DEBUG] 16:25:19.419 terminal.go:Getenv:161 → /home/pascal/.config/mytheme.omp.json
[TRACE] 16:25:19.419 terminal.go:Getenv(POSH_THEME) - 4.763µs
[DEBUG] 16:25:19.419 terminal.go:ResolveConfigPath:93 → config set using POSH_THEME: /home/pascal/.config/mytheme.omp.json
[TRACE] 16:25:19.419 terminal.go:ResolveConfigPath() - 38.466µs
[TRACE] 16:25:19.419 terminal.go:Init() - 781.675µs
[TRACE] 16:25:19.419 terminal.go:Shell() - 182ns
[TRACE] 16:25:19.419 terminal.go:Flags() - 94ns
[TRACE] 16:25:19.421 load.go:loadConfig() - 1.511474ms
[TRACE] 16:25:19.421 terminal.go:Flags() - 180ns
[DEBUG] 16:25:19.421 debug.go:59 → terminal program: Unknown
[DEBUG] 16:25:19.421 debug.go:59 → terminal shell: bash
[DEBUG] 16:25:19.421 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.421 terminal.go:Getenv(OMP_CACHE_DISABLED) - 5.253µs
[TRACE] 16:25:19.421 colors.go:SetAccentColor() - 1.468µs
[TRACE] 16:25:19.421 terminal.go:Shell() - 113ns
[DEBUG] 16:25:19.421 terminal.go:Getenv:161 → 5.2.32(1)-release
[TRACE] 16:25:19.421 terminal.go:Getenv(POSH_SHELL_VERSION) - 6.273µs
[DEBUG] 16:25:19.421 debug.go:PrintDebug:23 → segment: Title
[TRACE] 16:25:19.421 terminal_unix.go:Root() - 1.304µs
[TRACE] 16:25:19.421 terminal.go:Shell() - 120ns
[TRACE] 16:25:19.421 terminal.go:StatusCodes() - 113ns
[DEBUG] 16:25:19.421 terminal_unix.go:IsWsl:35 → false
[TRACE] 16:25:19.421 terminal_unix.go:IsWsl() - 11.423µs
[DEBUG] 16:25:19.421 terminal_unix.go:IsWsl:35 → false
[TRACE] 16:25:19.421 terminal_unix.go:IsWsl() - 4.677µs
[TRACE] 16:25:19.421 terminal.go:GOOS() - 83ns
[TRACE] 16:25:19.421 terminal.go:GOOS() - 62ns
[TRACE] 16:25:19.421 terminal.go:GOOS() - 72ns
[TRACE] 16:25:19.421 terminal.go:GOOS() - 66ns
[TRACE] 16:25:19.421 terminal.go:GOOS() - 65ns
[DEBUG] 16:25:19.421 terminal.go:User:331 → pascal
[TRACE] 16:25:19.421 terminal.go:User() - 4.515µs
[DEBUG] 16:25:19.422 terminal.go:Host:348 → pascal-laptop
[TRACE] 16:25:19.422 terminal.go:Host() - 6.268µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 69ns
[DEBUG] 16:25:19.422 terminal_unix.go:Platform:95 → fedora
[DEBUG] 16:25:19.422 terminal.go:Getenv:161 → 1
[TRACE] 16:25:19.422 terminal.go:Getenv(SHLVL) - 5.246µs
[TRACE] 16:25:19.422 terminal.go:TemplateCache() - 129.978µs
[TRACE] 16:25:19.422 text.go:Render({{ .PWD }}) - 580.586µs
[TRACE] 16:25:19.422 terminal.go:Flags() - 52ns
[DEBUG] 16:25:19.422 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.422 terminal.go:Getenv(POSH_CURSOR_LINE) - 3.973µs
[DEBUG] 16:25:19.422 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.422 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 3.27µs
[TRACE] 16:25:19.422 terminal.go:Flags() - 38ns
[TRACE] 16:25:19.422 terminal.go:Flags() - 41ns
[TRACE] 16:25:19.422 terminal.go:Flags() - 52ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 167ns
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Status
[TRACE] 16:25:19.422 terminal.go:Flags() - 217ns
[TRACE] 16:25:19.422 terminal.go:StatusCodes() - 120ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 144ns
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Path
[DEBUG] 16:25:19.422 status.go:formatStatus:45 → {{ .Code }}
[DEBUG] 16:25:19.422 path.go:145 → display_cygpath: false
[TRACE] 16:25:19.422 terminal.go:GOOS() - 115ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 63ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 6.109µs
[TRACE] 16:25:19.422 terminal.go:Flags() - 308ns
[TRACE] 16:25:19.422 terminal.go:Shell() - 129ns
[TRACE] 16:25:19.422 text.go:Render({{ .Code }}) - 63.637µs
[TRACE] 16:25:19.422 terminal.go:Shell() - 66ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 62ns
[DEBUG] 16:25:19.422 status.go:Enabled:37 → always_enabled: true
[TRACE] 16:25:19.422 terminal.go:TemplateCache() - 291ns
[DEBUG] 16:25:19.422 path.go:setMappedLocations:552 → mapped_locations_enabled: true
[DEBUG] 16:25:19.422 path.go:setMappedLocations:553 → 
[TRACE] 16:25:19.422 terminal.go:Flags() - 84ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 181ns
[DEBUG] 16:25:19.422 path.go:setMappedLocations:554 → 
[TRACE] 16:25:19.422 terminal.go:GOOS() - 81ns
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Svn
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Root
[DEBUG] 16:25:19.422 terminal_unix.go:IsWsl:35 → false
[TRACE] 16:25:19.422 terminal_unix.go:IsWsl() - 9.227µs
[TRACE] 16:25:19.422 terminal_unix.go:Root() - 1.924µs
[TRACE] 16:25:19.422 terminal_unix.go:IsWsl2() - 17.188µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 77ns
[TRACE] 16:25:19.422 terminal.go:Flags() - 107ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 80ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 55.676µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 118ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 102ns
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Session
[DEBUG] 16:25:19.422 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.422 terminal.go:Getenv(SSH_CONNECTION) - 6.779µs
[DEBUG] 16:25:19.422 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.422 terminal.go:CommandPath(svn) - 78.467µs
[TRACE] 16:25:19.422 terminal.go:Getenv(SSH_CLIENT) - 6.185µs
[TRACE] 16:25:19.422 terminal.go:HasCommand(svn) - 86.665µs
[DEBUG] 16:25:19.422 terminal_unix.go:Platform:95 → fedora
[TRACE] 16:25:19.422 terminal.go:GOOS() - 191ns
[ERROR] 16:25:19.422 terminal.go:HasParentFilePath:559 → stat .svn: no such file or directory
[TRACE] 16:25:19.422 terminal.go:GOOS() - 86ns
[TRACE] 16:25:19.422 terminal.go:HasParentFilePath(.svn) - 56.367µs
[DEBUG] 16:25:19.422 path.go:setMappedLocations:555 → ~
[TRACE] 16:25:19.422 terminal.go:Flags() - 117ns
[DEBUG] 16:25:19.422 path.go:setMappedLocations:560 → mapped_locations: map[/home/pascal/dev:  /home/pascal/tools/wireshark:󱙴  /mnt/data/dev:  /mnt/data/tools/wireshark:󱙴 ]
[TRACE] 16:25:19.422 text.go:Render(/home/pascal/tools/wireshark) - 610ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 141ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 341ns
[DEBUG] 16:25:19.422 segment.go:Execute:108 → segment: Git
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 8.485µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 130ns
[DEBUG] 16:25:19.422 terminal_unix.go:IsWsl:35 → false
[TRACE] 16:25:19.422 terminal_unix.go:IsWsl() - 8.958µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 89ns
[TRACE] 16:25:19.422 terminal_unix.go:IsWsl2() - 16.551µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 69ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 67ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 60ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 5.447µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 102ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 100ns
[DEBUG] 16:25:19.422 terminal.go:CommandPath:395 → /usr/bin/git
[TRACE] 16:25:19.422 terminal.go:GOOS() - 45ns
[TRACE] 16:25:19.422 terminal.go:CommandPath(git) - 49.799µs
[TRACE] 16:25:19.422 text.go:Render(/mnt/data/dev) - 543ns
[TRACE] 16:25:19.422 terminal.go:HasCommand(git) - 57.511µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 70ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 6.597µs
[DEBUG] 16:25:19.422 git.go:shouldDisplay:326 → fetch_bare_info: false
[TRACE] 16:25:19.422 terminal.go:GOOS() - 83ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 117ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 98ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 101ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 20.157µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 319ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 65ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 301ns
[TRACE] 16:25:19.422 text.go:Render(/mnt/data/tools/wireshark) - 739ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 87ns
[TRACE] 16:25:19.422 terminal.go:PathSeparator() - 9.461µs
[TRACE] 16:25:19.422 terminal.go:GOOS() - 73ns
[TRACE] 16:25:19.422 terminal.go:GOOS() - 89ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 91ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 210ns
[TRACE] 16:25:19.423 terminal.go:PathSeparator() - 27.012µs
[TRACE] 16:25:19.423 terminal.go:GOOS() - 101ns
[TRACE] 16:25:19.423 terminal.go:Flags() - 211ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 162ns
[DEBUG] 16:25:19.423 segment.go:Execute:108 → segment: Os
[TRACE] 16:25:19.423 terminal.go:GOOS() - 65ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 107ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 71ns
[DEBUG] 16:25:19.423 terminal_unix.go:Platform:95 → fedora
[TRACE] 16:25:19.423 text.go:Render(/home/pascal/dev) - 602ns
[DEBUG] 16:25:19.423 os.go:Enabled:38 → display_distro_name: false
[TRACE] 16:25:19.423 terminal.go:GOOS() - 66ns
[TRACE] 16:25:19.423 terminal.go:PathSeparator() - 7.972µs
[TRACE] 16:25:19.423 terminal.go:GOOS() - 95ns
[DEBUG] 16:25:19.423 os.go:getDistroIcon:83 → 
[TRACE] 16:25:19.423 terminal.go:TemplateCache() - 232ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 100ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 72ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 88ns
[TRACE] 16:25:19.423 terminal.go:PathSeparator() - 6.596µs
[TRACE] 16:25:19.423 terminal.go:GOOS() - 62ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 61ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 64ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 224ns
[TRACE] 16:25:19.423 terminal.go:PathSeparator() - 8.496µs
[TRACE] 16:25:19.423 terminal.go:GOOS() - 50ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 85ns
[TRACE] 16:25:19.423 text.go:Render( {{ if .WSL }}WSL at {{ end }}{{.Icon}} ) - 108.76µs
[TRACE] 16:25:19.423 terminal.go:GOOS() - 68ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 93ns
[TRACE] 16:25:19.423 terminal.go:PathSeparator() - 6.161µs
[TRACE] 16:25:19.423 text.go:Render(diamond) - 508ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 67ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 52ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 103ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 58ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 46ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 105ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 62ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 66ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 93ns
[TRACE] 16:25:19.423 terminal.go:GOOS() - 102ns
[DEBUG] 16:25:19.423 path.go:makeFolderFormatMap:827 → NO DATA
[TRACE] 16:25:19.423 terminal.go:GOOS() - 113ns
[DEBUG] 16:25:19.424 git.go:getSwitchMode:909 → ignore_submodules: map[]
[DEBUG] 16:25:19.424 path.go:setStyle:209 → agnoster_full
[DEBUG] 16:25:19.424 path.go:colorizePath:728 → cycle: []
[DEBUG] 16:25:19.424 path.go:getFolderSeparator:263 → NO DATA
[DEBUG] 16:25:19.424 path.go:getFolderSeparator:265 → /
[TRACE] 16:25:19.424 terminal.go:TemplateCache() - 181ns
[DEBUG] 16:25:19.424 path.go:colorizePath:731 → cycle_folder_separator: false
[DEBUG] 16:25:19.424 path.go:colorizePath:732 → %s
[DEBUG] 16:25:19.424 path.go:colorizePath:734 → %s
[DEBUG] 16:25:19.424 path.go:colorizePath:735 → %s
[DEBUG] 16:25:19.424 path.go:colorizePath:736 → %s
[TRACE] 16:25:19.424 terminal.go:GOOS() - 212ns
[TRACE] 16:25:19.424 terminal.go:TemplateCache() - 203ns
[TRACE] 16:25:19.424 terminal.go:GOOS() - 67ns
[TRACE] 16:25:19.424 terminal.go:StackCount() - 135ns
[TRACE] 16:25:19.424 terminal_unix.go:DirIsWritable(/home/pascal/tools/wireshark/git) - 9.374µs
[TRACE] 16:25:19.424 terminal.go:TemplateCache() - 78ns
[DEBUG] 16:25:19.425 terminal.go:HTTPRequest:499 ↓
    GET /repos/jandedobbeleer/oh-my-posh/releases/latest HTTP/1.1
    Host: api.github.com
    User-Agent: Go-http-client/1.1
    Accept-Encoding: gzip
    
    
[TRACE] 16:25:19.426 terminal.go:TemplateCache() - 495ns
[TRACE] 16:25:19.426 text.go:Render( {{ if .SSHSession }} {{ end }}{{ .UserName }} ) - 2.118499ms
[TRACE] 16:25:19.426 text.go:Render(diamond) - 354ns
[DEBUG] 16:25:19.454 terminal.go:RunCommand:371 ↓
    # branch.oid c9b894b873c6aa82b66084487fc627d76ed7e496
    # branch.head master
    # branch.upstream upstream/master
    # branch.ab +0 -0
    ? toto
[TRACE] 16:25:19.454 terminal.go:RunCommand(git -C /home/pascal/tools/wireshark/git --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 30.210827ms
[DEBUG] 16:25:19.454 git.go:setGitHEADContext:669 → 
[DEBUG] 16:25:19.454 scm.go:formatBranch:106 → mapped_branches: map[]
[DEBUG] 16:25:19.454 scm.go:formatBranch:122 → full_branch_path: true
[DEBUG] 16:25:19.454 scm.go:formatBranch:128 → branch_max_length: 25
[DEBUG] 16:25:19.454 terminal.go:HasFolder:272 → false
[TRACE] 16:25:19.454 terminal.go:HasFolder(/home/pascal/tools/wireshark/git/.git/rebase-merge) - 12.284µs
[DEBUG] 16:25:19.454 terminal.go:HasFolder:272 → false
[TRACE] 16:25:19.454 terminal.go:HasFolder(/home/pascal/tools/wireshark/git/.git/rebase-apply) - 5.672µs
[DEBUG] 16:25:19.454 git.go:setGitHEADContext:720 → 
[DEBUG] 16:25:19.454 terminal.go:HasFilesInDir:243 → false
[TRACE] 16:25:19.454 terminal.go:HasFilesInDir(MERGE_MSG) - 61.57µs
[DEBUG] 16:25:19.454 terminal.go:HasFilesInDir:243 → false
[TRACE] 16:25:19.454 terminal.go:HasFilesInDir(CHERRY_PICK_HEAD) - 7.649µs
[DEBUG] 16:25:19.454 terminal.go:HasFilesInDir:243 → false
[TRACE] 16:25:19.454 terminal.go:HasFilesInDir(REVERT_HEAD) - 6.744µs
[DEBUG] 16:25:19.454 terminal.go:HasFilesInDir:243 → false
[TRACE] 16:25:19.454 terminal.go:HasFilesInDir(sequencer/todo) - 5.873µs
[DEBUG] 16:25:19.454 git.go:483 → ≡
[DEBUG] 16:25:19.454 git.go:Enabled:205 → fetch_upstream_icon: true
[DEBUG] 16:25:19.455 git.go:getUpstreamIcon:547 → upstream_icons: map[]
[DEBUG] 16:25:19.455 git.go:getUpstreamIcon:568 →  
[TRACE] 16:25:19.455 terminal.go:TemplateCache() - 363ns
[ERROR] 16:25:19.526 terminal.go:HTTPRequest:512 → HTTP status code 403
[TRACE] 16:25:19.526 terminal.go:HTTPRequest(https://api.github.com/repos/jandedobbeleer/oh-my-posh/releases/latest) - 102.08743ms
[TRACE] 16:25:19.526 text.go:Render(powerline) - 1.422µs
[TRACE] 16:25:19.526 terminal.go:TemplateCache() - 2.74µs
[TRACE] 16:25:19.526 text.go:Render(  {{ path .Path .Location }} ) - 305.506µs
[TRACE] 16:25:19.526 text.go:Render(powerline) - 231ns
[TRACE] 16:25:19.526 text.go:Render( {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }} {{ end }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }}  {{ .Working.String }}{{ end }}{{ if .Staging.Changed }}  {{ .Staging.String }}{{ end }} ) - 159.341µs
[TRACE] 16:25:19.526 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 40.847µs
[TRACE] 16:25:19.526 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 38.878µs
[TRACE] 16:25:19.526 text.go:Render(powerline) - 213ns
[TRACE] 16:25:19.526 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 31.04µs
[TRACE] 16:25:19.527 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 30.17µs
[TRACE] 16:25:19.527 text.go:Render(powerline) - 176ns
[TRACE] 16:25:19.527 text.go:Render(powerline) - 201ns
[TRACE] 16:25:19.527 text.go:Render( {{ if gt .Code 0 }} {{ reason .Code }}{{ else }}{{ end }} ) - 33.965µs
[TRACE] 16:25:19.527 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 17.772µs
[TRACE] 16:25:19.527 text.go:Render(diamond) - 168ns
[TRACE] 16:25:19.527 text.go:Render({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 31.638µs
[TRACE] 16:25:19.527 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 17.922µs
[TRACE] 16:25:19.527 terminal.go:Shell() - 207ns
[TRACE] 16:25:19.527 terminal.go:Shell() - 62ns
[TRACE] 16:25:19.527 terminal.go:Flags() - 60ns
[TRACE] 16:25:19.527 terminal.go:GOOS() - 73ns
[TRACE] 16:25:19.527 terminal.go:Flags() - 116ns
[DEBUG] 16:25:19.527 segment.go:Execute:108 → segment: Sysinfo
[TRACE] 16:25:19.527 terminal.go:Flags() - 127ns
[TRACE] 16:25:19.527 terminal.go:GOOS() - 84ns
[DEBUG] 16:25:19.527 sysinfo.go:Enabled:25 → precision: 2
[TRACE] 16:25:19.527 terminal.go:GOOS() - 65ns
[DEBUG] 16:25:19.527 segment.go:Execute:108 → segment: Executiontime
[DEBUG] 16:25:19.527 segment.go:Execute:108 → segment: Battery
[DEBUG] 16:25:19.527 executiontime.go:Enabled:55 → always_enabled: false
[DEBUG] 16:25:19.527 terminal_unix.go:IsWsl:35 → false
[TRACE] 16:25:19.527 terminal.go:ExecutionTime() - 95ns
[TRACE] 16:25:19.527 terminal_unix.go:IsWsl() - 6.323µs
[DEBUG] 16:25:19.527 executiontime.go:Enabled:57 → threshold: 500.000000
[TRACE] 16:25:19.527 text.go:Render(plain) - 452ns
[TRACE] 16:25:19.528 terminal_windows_nix.go:BatteryState() - 895.037µs
[DEBUG] 16:25:19.528 battery.go:Enabled:53 →  
[TRACE] 16:25:19.528 terminal.go:TemplateCache() - 234ns
[TRACE] 16:25:19.528 terminal.go:TemplateCache() - 175ns
[TRACE] 16:25:19.528 text.go:Render( {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }} ) - 190.433µs
[TRACE] 16:25:19.528 text.go:Render({{if eq "Charging" .State.String}}#40c4ff{{end}}) - 38.925µs
[TRACE] 16:25:19.528 text.go:Render({{if eq "Discharging" .State.String}}#ff5722{{end}}) - 18.943µs
[TRACE] 16:25:19.528 text.go:Render({{if eq "Full" .State.String}}#4caf50{{end}}) - 181.473µs
[TRACE] 16:25:19.528 text.go:Render(plain) - 471ns
[TRACE] 16:25:19.528 terminal.go:TemplateCache() - 198ns
[TRACE] 16:25:19.528 text.go:Render({{if eq "Charging" .State.String}}#40c4ff{{end}}) - 64.361µs
[TRACE] 16:25:19.528 text.go:Render({{if eq "Discharging" .State.String}}#ff5722{{end}}) - 42.871µs
[TRACE] 16:25:19.528 text.go:Render({{if eq "Full" .State.String}}#4caf50{{end}}) - 39.098µs
[TRACE] 16:25:19.528 text.go:Render(  {{ round .PhysicalPercentUsed .Precision }}% ) - 54.484µs
[TRACE] 16:25:19.528 text.go:Render(plain) - 265ns
[DEBUG] 16:25:19.529 terminal_unix.go:TerminalWidth:88 → terminal width: 238
[TRACE] 16:25:19.529 terminal_unix.go:TerminalWidth() - 1.034501ms
[TRACE] 16:25:19.529 terminal.go:Shell() - 212ns
[TRACE] 16:25:19.529 terminal.go:Shell() - 44ns
[TRACE] 16:25:19.530 terminal.go:Flags() - 90ns
[TRACE] 16:25:19.530 terminal.go:GOOS() - 77ns
[DEBUG] 16:25:19.530 segment.go:Execute:108 → segment: Status
[TRACE] 16:25:19.530 terminal.go:StatusCodes() - 97ns
[DEBUG] 16:25:19.530 status.go:formatStatus:45 → {{ .Code }}
[TRACE] 16:25:19.530 terminal.go:Flags() - 174ns
[TRACE] 16:25:19.530 terminal.go:GOOS() - 85ns
[TRACE] 16:25:19.530 text.go:Render({{ .Code }}) - 40.628µs
[DEBUG] 16:25:19.530 segment.go:Execute:108 → segment: Text
[DEBUG] 16:25:19.530 status.go:Enabled:37 → always_enabled: true
[TRACE] 16:25:19.530 terminal.go:TemplateCache() - 170ns
[TRACE] 16:25:19.530 terminal.go:TemplateCache() - 162ns
[TRACE] 16:25:19.530 text.go:Render(╰─) - 365ns
[TRACE] 16:25:19.530 text.go:Render(plain) - 194ns
[TRACE] 16:25:19.530 text.go:Render() - 167ns
[TRACE] 16:25:19.530 terminal.go:TemplateCache() - 60ns
[TRACE] 16:25:19.530 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 134.735µs
[TRACE] 16:25:19.530 text.go:Render(plain) - 109ns
[TRACE] 16:25:19.530 text.go:Render({{ if gt .Code 0 }}p:red{{ end }}) - 14.725µs
[TRACE] 16:25:19.530 terminal.go:Shell() - 119ns
[TRACE] 16:25:19.530 terminal.go:Shell() - 32ns
[TRACE] 16:25:19.530 terminal.go:Flags() - 40ns
[TRACE] 16:25:19.530 terminal.go:Shell() - 26ns
[DEBUG] 16:25:19.530 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.530 terminal.go:Getenv(LOCALAPPDATA) - 3.455µs
[DEBUG] 16:25:19.530 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.530 terminal.go:Getenv(OMP_CACHE_DIR) - 2.293µs
[DEBUG] 16:25:19.530 terminal.go:Getenv:161 → NO DATA
[TRACE] 16:25:19.530 terminal.go:Getenv(XDG_CACHE_HOME) - 2.172µs
[TRACE] 16:25:19.530 terminal.go:CachePath() - 31.903µs
[TRACE] 16:25:19.530 terminal.go:Flags() - 31ns
```

--- a/src/config/migrate.go
+++ b/src/config/migrate.go
@@ -30,10 +30,16 @@ func (segment *Segment) migrate(version int) {
 
 	// Cache settings
 	delete(segment.Properties, "cache_version")
-	segment.Cache = segment.migrateCache()
 
 	segment.IncludeFolders = segment.migrateFolders(includeFolders)
 	segment.ExcludeFolders = segment.migrateFolders(excludeFolders)
+
+	switch segment.Type { //nolint:exhaustive
+	case UPGRADE:
+		segment.timeoutToDuration()
+	default:
+		segment.timeoutToCache()
+	}
 }
 
 func (segment *Segment) hasProperty(property properties.Property) bool {
@@ -45,22 +51,33 @@ func (segment *Segment) hasProperty(property properties.Property) bool {
 	return false
 }
 
-func (segment *Segment) migrateCache() *cache.Config {
+func (segment *Segment) timeoutToCache() {
 	if !segment.hasProperty(cacheTimeout) {
-		return nil
+		return
 	}
 
 	timeout := segment.Properties.GetInt(cacheTimeout, 0)
 	delete(segment.Properties, cacheTimeout)
 
 	if timeout == 0 {
-		return nil
+		return
 	}
 
-	return &cache.Config{
+	segment.Cache = &cache.Config{
 		Duration: cache.ToDuration(timeout * 60),
 		Strategy: cache.Folder,
 	}
+}
+
+func (segment *Segment) timeoutToDuration() {
+	timeout := segment.Properties.GetInt(cacheTimeout, 0)
+	delete(segment.Properties, cacheTimeout)
+
+	if timeout == 0 {
+		return
+	}
+
+	segment.Properties[properties.CacheDuration] = cache.ToDuration(timeout * 60)
 }
 
 func (segment *Segment) migrateFolders(property properties.Property) []string {

--- a/src/config/migrate_test.go
+++ b/src/config/migrate_test.go
@@ -9,7 +9,7 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestMigrateCache(t *testing.T) {
+func TestTimeoutToCache(t *testing.T) {
 	cases := []struct {
 		Expected *cache.Config
 		Case     string
@@ -27,8 +27,8 @@ func TestMigrateCache(t *testing.T) {
 			},
 		}
 
-		got := segment.migrateCache()
-		assert.Equal(t, tc.Expected, got, tc.Case)
+		segment.timeoutToCache()
+		assert.Equal(t, tc.Expected, segment.Cache, tc.Case)
 	}
 }
 

--- a/src/properties/map.go
+++ b/src/properties/map.go
@@ -45,6 +45,8 @@ const (
 	DefaultHTTPTimeout = 20
 	// Files to trigger the segment on
 	Files Property = "files"
+	// Duration of the cache
+	CacheDuration Property = "cache_duration"
 )
 
 type Map map[Property]any

--- a/src/segments/upgrade.go
+++ b/src/segments/upgrade.go
@@ -1,11 +1,16 @@
 package segments
 
 import (
+	"encoding/json"
+	"errors"
+
 	"github.com/jandedobbeleer/oh-my-posh/src/build"
+	"github.com/jandedobbeleer/oh-my-posh/src/cache"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
 	"github.com/jandedobbeleer/oh-my-posh/src/upgrade"
 )
 
-type upgradeData struct {
+type UpgradeCache struct {
 	Latest  string `json:"latest"`
 	Current string `json:"current"`
 }
@@ -16,8 +21,12 @@ type Upgrade struct {
 	// deprecated
 	Version string
 
-	upgradeData
+	UpgradeCache
 }
+
+const (
+	UPGRADECACHEKEY = "upgrade_segment"
+)
 
 func (u *Upgrade) Template() string {
 	return " \uf019 "
@@ -25,27 +34,58 @@ func (u *Upgrade) Template() string {
 
 func (u *Upgrade) Enabled() bool {
 	u.Current = build.Version
-
-	latest, err := u.checkUpdate(u.Current)
+	latest, err := u.cachedLatest(u.Current)
+	if err != nil {
+		latest, err = u.checkUpdate(u.Current)
+	}
 
 	if err != nil || u.Current == latest.Latest {
 		return false
 	}
 
-	u.upgradeData = *latest
+	u.UpgradeCache = *latest
 	u.Version = u.Latest
 	return true
 }
 
-func (u *Upgrade) checkUpdate(current string) (*upgradeData, error) {
+func (u *Upgrade) cachedLatest(current string) (*UpgradeCache, error) {
+	data, ok := u.env.Cache().Get(UPGRADECACHEKEY)
+	if !ok {
+		return nil, errors.New("no cache data")
+	}
+
+	var cacheJSON UpgradeCache
+	err := json.Unmarshal([]byte(data), &cacheJSON)
+	if err != nil {
+		return nil, err // invalid cache data
+	}
+
+	if current != cacheJSON.Current {
+		return nil, errors.New("version changed, run the check again")
+	}
+
+	return &cacheJSON, nil
+}
+
+func (u *Upgrade) checkUpdate(current string) (*UpgradeCache, error) {
 	tag, err := upgrade.Latest(u.env)
 	if err != nil {
 		return nil, err
 	}
 
-	return &upgradeData{
-		// strip leading v
-		Latest:  tag[1:],
+	latest := tag[1:]
+	cacheData := &UpgradeCache{
+		Latest:  latest,
 		Current: current,
-	}, nil
+	}
+	cacheJSON, err := json.Marshal(cacheData)
+	if err != nil {
+		return nil, err
+	}
+
+	// update cache
+	duration := u.props.GetString(properties.CacheDuration, string(cache.ONEWEEK))
+	u.env.Cache().Set(UPGRADECACHEKEY, string(cacheJSON), cache.Duration(duration))
+
+	return cacheData, nil
 }

--- a/src/segments/upgrade_test.go
+++ b/src/segments/upgrade_test.go
@@ -6,11 +6,13 @@ import (
 	"testing"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/build"
+	cache_ "github.com/jandedobbeleer/oh-my-posh/src/cache/mock"
 	"github.com/jandedobbeleer/oh-my-posh/src/properties"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime/mock"
 	"github.com/jandedobbeleer/oh-my-posh/src/upgrade"
 
 	"github.com/alecthomas/assert"
+	testify_ "github.com/stretchr/testify/mock"
 )
 
 func TestUpgrade(t *testing.T) {
@@ -39,20 +41,41 @@ func TestUpgrade(t *testing.T) {
 			Error: errors.New("error"),
 		},
 		{
+			Case:            "On previous, from cache",
+			HasCache:        true,
+			CurrentVersion:  "1.0.2",
+			LatestVersion:   "1.0.3",
+			CachedVersion:   "1.0.2",
+			ExpectedEnabled: true,
+		},
+		{
 			Case:           "On latest, version changed",
+			HasCache:       true,
 			CurrentVersion: "1.0.2",
 			LatestVersion:  "1.0.2",
+			CachedVersion:  "1.0.1",
 		},
 		{
 			Case:            "On previous, version changed",
+			HasCache:        true,
 			CurrentVersion:  "1.0.2",
 			LatestVersion:   "1.0.3",
+			CachedVersion:   "1.0.1",
 			ExpectedEnabled: true,
 		},
 	}
 
 	for _, tc := range cases {
 		env := new(mock.Environment)
+		cache := &cache_.Cache{}
+
+		env.On("Cache").Return(cache)
+		if len(tc.CachedVersion) == 0 {
+			tc.CachedVersion = tc.CurrentVersion
+		}
+		cacheData := fmt.Sprintf(`{"latest":"%s", "current": "%s"}`, tc.LatestVersion, tc.CachedVersion)
+		cache.On("Get", UPGRADECACHEKEY).Return(cacheData, tc.HasCache)
+		cache.On("Set", testify_.Anything, testify_.Anything, testify_.Anything)
 
 		build.Version = tc.CurrentVersion
 

--- a/website/docs/configuration/segment.mdx
+++ b/website/docs/configuration/segment.mdx
@@ -22,9 +22,7 @@ understand how to configure a segment.
             foreground: "#ffffff",
             background: "#61AFEF",
             template: " {{ .Path }} ",
-            properties: {
-              include_folders: ["/Users/posh/Projects"],
-            },
+            include_folders: ["/Users/posh/Projects"],
           },
         ],
       },

--- a/website/docs/segments/system/upgrade.mdx
+++ b/website/docs/segments/system/upgrade.mdx
@@ -18,8 +18,17 @@ import Config from "@site/src/components/Config.js";
     style: "plain",
     foreground: "#111111",
     background: "#FFD664",
+    properties: {
+      cache_duration: "168h",
+    },
   }}
 />
+
+## Properties
+
+| Name             |   Type   | Default | Description                                                                                                                                                                                     |
+| ---------------- | :------: | :-----: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `cache_duration` | `string` |   ``    | the duration for which the segment will be cached. The duration is a string in the format `1h2m3s`. The duration is parsed using the [time.ParseDuration] function from the Go standard library |
 
 ## Template ([info][templates])
 
@@ -39,3 +48,4 @@ import Config from "@site/src/components/Config.js";
 | `.Latest`  | `string` | the latest available version number |
 
 [templates]: /docs/configuration/templates
+[time.ParseDuration]: https://golang.org/pkg/time/#ParseDuration

